packages/repository/src/repositories/relation.factory.ts

```diff
@@ -4,21 +4,29 @@
 // License text available at https://opensource.org/licenses/MIT
 
 import {EntityCrudRepository} from './repository';
-import {RelationType} from '../decorators/relation.decorator';
+import {RelationType, RELATIONS_KEY} from '../decorators/relation.decorator';
 import {Entity} from '../model';
 import {
   HasManyEntityCrudRepository,
   DefaultHasManyEntityCrudRepository,
 } from './relation.repository';
+import {DefaultCrudRepository} from './legacy-juggler-bridge';
+import {MetadataInspector} from '@loopback/context';
 
 export interface RelationDefinitionBase {
   type: RelationType;
-  keyTo: string;
 }
 
 export interface HasManyDefinition extends RelationDefinitionBase {
   type: RelationType.hasMany;
+  keyTo: string;
+  keyFrom: string;
 }
+
+export type constrainRepositoryFunction<T extends Entity> = (
+  key: Partial<T>,
+) => HasManyEntityCrudRepository<T>;
```

Comment:
I find this name a bit too long. Let's discuss if we there is room for making it shorter.

1. Are we expecting repositories for HasMany relation that are not CRUD? If not, then remove `Crud` from the name.
2. Is it possible to implement HasMany relation for models that are not inheriting from Entity (don't have a primary key e.g. `id`)? If not, then remove `Entity` from the name.

I think (but may be wrong!) that answers to both my questions are true, in which case I am proposing the following name: `HasManyRepository<T>`.

@b-admike @shimks @raymondfeng thoughts?

Reply:
These are great questions to ask for the future when naming interfaces!

>Are we expecting repositories for HasMany relation that are not CRUD? If not, then remove Crud from the name.

No we're not; sounds good.

>Is it possible to implement HasMany relation for models that are not inheriting from Entity (don't have a primary key e.g. id)? If not, then remove Entity from the name.

No I don't think it is; will remove.